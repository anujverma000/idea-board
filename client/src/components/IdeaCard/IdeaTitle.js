import React, { useState } from "react";
import styled from "styled-components";

const IdeaTitle = ({ idea, saveIdea }) => {

  const maxLength = 140;

  const [ ideaTitle, setIdeaTitle ] = useState(idea.title);
  const [ isEdited, setIsEdited ] = useState(false);

  const saveIdeaUpdates = event => {
    if (event.keyCode === 13) {
      updateIdea();
    };
  }

  const updateIdea = () => {
    if (isEdited) {
      saveIdea(getUpdatedIdea());
      setIsEdited(false);
    }
  }

  const handleChange = e => {
    setIdeaTitle(e.target.value);
    setIsEdited(true);
  }

  const getUpdatedIdea = () => ({...idea, title: ideaTitle})

  return <TitleEditable
            value={ideaTitle} 
            onChange={handleChange} 
            type="text"
            maxLength={maxLength}
            autoFocus={ idea.new ? true: false }
            onKeyUp={saveIdeaUpdates}
            onBlur={updateIdea}
            placeholder="Add Idea"/>
}
/**
 *  STYLES
 */

const TitleEditable = styled.input`
  color: ${props => props.theme.textColorStrong};
  font-size: 16px;
  font-weight: bold;
  width: 95%;
  padding: 8px;
  border-radius: 4px;
  border: none;
  &:focus{
    outline: 1px solid ${props => props.theme.primaryColor};
  }
`
export default IdeaTitle;