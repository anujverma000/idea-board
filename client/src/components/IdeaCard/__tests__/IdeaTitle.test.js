import React from 'react';
import renderer from 'react-test-renderer';
import IdeaTitle from '../IdeaTitle';

it('renders correctly', () => {
  const testIdea = {
    id: '123',
    title: 'Test Idea Title',
    body: 'Test Idea Body',
    created_date: 'Aug 2, 2019'
  };
  const tree = renderer.create(<IdeaTitle idea={testIdea}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
