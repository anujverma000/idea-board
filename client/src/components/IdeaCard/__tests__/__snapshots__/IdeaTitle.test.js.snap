// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  autoFocus={false}
  className="sc-bdVaJa eTVgRM"
  maxLength={140}
  onBlur={[Function]}
  onChange={[Function]}
  onKeyUp={[Function]}
  placeholder="Add Idea"
  type="text"
  value="Test Idea Title"
/>
`;
